### PR TITLE
Sass refactoring for 3.2

### PR DESCRIPTION
**Pull request by vxsx**
## General changes info
All `cms_` classnames were changed to `cms-` classnames. 
Compass switched to libsass (through gulp frontend-tooling)
Icons are now with iconfont automatically generated from svgs
There are now a lot more variables which allow for easy changes in the cms look and feel without digging too much in the cms code.

All the work was done in separate PRs listed below.

## Infrastructure updates
- Dropping ruby/compass in favour of libsass https://github.com/FinalAngel/django-cms/pull/6
- Runing csscomb over sass to enforce coding guidelines https://github.com/FinalAngel/django-cms/pull/7
- Removing redundant vendor prefix mixins in favor of autoprefixer https://github.com/FinalAngel/django-cms/pull/8
- Add documentation about new Sass setup https://github.com/FinalAngel/django-cms/pull/9
- Ability to have sourcemaps for Sass during development https://github.com/FinalAngel/django-cms/pull/30
- Add system to create iconfonts https://github.com/FinalAngel/django-cms/pull/10/

## Migrating icons
- Test: https://github.com/FinalAngel/django-cms/pull/14
- Docs: https://github.com/FinalAngel/django-cms/pull/15
- Modal: https://github.com/FinalAngel/django-cms/pull/16 & https://github.com/FinalAngel/django-cms/pull/17
- Sideframe: https://github.com/FinalAngel/django-cms/pull/18
- Toolbar: https://github.com/FinalAngel/django-cms/pull/26
- Structure board: https://github.com/FinalAngel/django-cms/pull/27
- Remaining: https://github.com/FinalAngel/django-cms/pull/26
- update for mixins: https://github.com/FinalAngel/django-cms/pull/31

## General fixes on css / templates
- Convert tabs to spaces https://github.com/FinalAngel/django-cms/pull/33
- tabs to spaces for HTML https://github.com/FinalAngel/django-cms/pull/36

## Adapt CSS to Aldryn Boilerplate standards
- cms.base https://github.com/FinalAngel/django-cms/pull/12
- components/common https://github.com/FinalAngel/django-cms/pull/19
- components/modal https://github.com/FinalAngel/django-cms/pull/20
- components/clipboard https://github.com/FinalAngel/django-cms/pull/21
- components/sideframe https://github.com/FinalAngel/django-cms/pull/22
- components/structureboard https://github.com/FinalAngel/django-cms/pull/23
- components/submenu https://github.com/FinalAngel/django-cms/pull/24
- components/toolbar https://github.com/FinalAngel/django-cms/pull/25
- components/content https://github.com/FinalAngel/django-cms/pull/29

## Adding more uniform variables across all components
- buttons, colors https://github.com/FinalAngel/django-cms/pull/34
- clipboard, zindex https://github.com/FinalAngel/django-cms/pull/37
- mixins https://github.com/FinalAngel/django-cms/pull/38
- dialog https://github.com/FinalAngel/django-cms/pull/39
- modal https://github.com/FinalAngel/django-cms/pull/40
- sideframe + a bit of clipboard https://github.com/FinalAngel/django-cms/pull/41
- structureboard https://github.com/FinalAngel/django-cms/pull/42
- submenu + tooltip + bits of structureboard and clipboard https://github.com/FinalAngel/django-cms/pull/43
- toolbar + things around toolbar https://github.com/FinalAngel/django-cms/pull/44
- remaining things https://github.com/FinalAngel/django-cms/pull/45
 
:boom: 
/cc @FinalAngel @yakky @mkoistinen

### CODE REVIEW

**Comment by FinalAngel** [comment]
:+1:

File: cms/static/cms/js/modules/cms.base.js [hunk 1]
```diff
@@ -12,231 +12,231 @@ window.Class = window.Class || undefined;
...
+                else {
+                    // ensure page is always reloaded #3413
+                    parent.location.reload()
+                }
+            }, timeout || 0);
+        },
+
+        // disable multiple form submissions
+        preventSubmit: function () {
+            var forms = $('#cms-toolbar').find('form');
```
Comment (darioalbanesi) [inline cms/static/cms/js/modules/cms.base.js]: why ID here? could we use .js-cms-toolbar?

File: cms/static/cms/js/modules/cms.base.js [hunk 2]
```diff
@@ -12,231 +12,231 @@ window.Class = window.Class || undefined;
...
+                beforeSend: function (xhr) {
+                    // set csrf_token
+                    xhr.setRequestHeader("X-CSRFToken", csrf_token);
+                }
+            });
+        },
+
+        // handles the tooltip for the plugins
+        showTooltip: function (name, id) {
+            var tooltip = $('.cms-tooltip');
```
Comment (darioalbanesi) [inline cms/static/cms/js/modules/cms.base.js]: here the same .js-cms-tooltip

File: cms/static/cms/js/modules/cms.base.js (hunk 1 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.base.js]: we could, but that would come later as part of js refactoring/cleanup

File: cms/static/cms/js/modules/cms.base.js (hunk 2 above)
Comment (vxsx) [inline cms/static/cms/js/modules/cms.base.js]: same answer

File: cms/static/cms/js/modules/cms.base.js (hunk 1 above)
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.base.js]: sadly not currently @darioalbanesi as there are various dependencies affected - but it's a goal

File: cms/static/cms/js/modules/cms.base.js (hunk 2 above)
Comment (FinalAngel) [inline cms/static/cms/js/modules/cms.base.js]: thats correct @vxsx

**Comment by yakky** [comment]
@FinalAngel @vxsx This is really awesome
My concern is about shipping gulpfile / packages.json in ``/static/`` as it's going to be collected and exposed by Django.
We can avoid this by removing those files (along with scss) from the packages uploaded to PyPi

**Comment by FinalAngel** [comment]
@yakky we can also move these files to the root if that is alright for you. We decided not to do this to avoid cluttering there. But we'd actually prefer it so you can execute the gulp commands from within there?

**Comment by vxsx** [comment]
@yakky it actually would be great if we could move package.json / gulpfile to the root
it's much more convenient to do 
```
cd django-cms
gulp
```
than 
```
cd django-cms/cms/static/cms
gulp
```

**Comment by yakky** [comment]
I'm definitely :+1: on moving in the root
It makes so much sense: the root is where every command and developer-side configuration is 
ping @mkoistinen  @evildmp

**Comment by vxsx** [comment]
@yakky updated the setup https://github.com/FinalAngel/django-cms/pull/50 and docs https://github.com/FinalAngel/django-cms/pull/51

**Comment by FinalAngel** [comment]
@yakky :+1:

**Comment by yakky** [comment]
@vxsx @FinalAngel let's go with this, then

**Pull request merged**